lnwallet/update_log.go

```diff
@@ -144,15 +146,14 @@ func compactLogs(ourLog, theirLog *updateLog,
 	localChainTail, remoteChainTail uint64) {
 
 	compactLog := func(logA, logB *updateLog) {
-		var nextA *list.Element
+		var nextA *fn.Node[*PaymentDescriptor]
 		for e := logA.Front(); e != nil; e = nextA {
 			// Assign next iteration element at top of loop because
 			// we may remove the current element from the list,
 			// which can change the iterated sequence.
 			nextA = e.Next()
 
-			//nolint:forcetypeassert
-			htlc := e.Value.(*PaymentDescriptor)
+			htlc := e.Value
```

Comment:
in hindsight I prolly would make a new method `Value()` or `Current()` so it's impossible to mutate the node's internal data.

Reply:
Yeah that's probably a good idea. I tried to copy the official golang linked list API exactly when making this but we can diverge them.

Reply:
This would be a breaking change to the fn.List API however and so should not be a blocker for this commit. I'll add it to the list of changes to make for `fn@v2.0`